zjit/src/cruby_methods.rs

```diff
@@ -247,6 +248,13 @@ fn inline_kernel_itself(_fun: &mut hir::Function, _block: hir::BlockId, recv: hi
     None
 }
 
+fn inline_kernel_block_given_p(fun: &mut hir::Function, block: hir::BlockId, _recv: hir::InsnId, args: &[hir::InsnId], _state: hir::InsnId) -> Option<hir::InsnId> {
+    let &[] = args else { return None; };
+    // TODO(max): In iseq body types that are not ISEQ_TYPE_METHOD, rewrite to Constant false.
```

Comment:
```suggestion
    // TODO(max): In local iseq types that are not ISEQ_TYPE_METHOD, rewrite to Constant false.
```

The test is essentially "is my iseq rooted in a method?"